### PR TITLE
Use common thread post id types

### PR DESCRIPTION
**Pull request by mnaamani**
Makes ThreadId and PostId types configurable on trait of Forum module, and configure both to be `u64`.

Avoids future issues similar to https://github.com/Joystream/apps/issues/434

### CODE REVIEW

**Comment by mnaamani** [comment]
I could have gone with renaming ThreadId and PostId like so:

`ForumThreadId, ForumPostId, ProposalDiscussionThreadId, .. `

however this approach seemed cleaner...thoughts?

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -94,6 +94,12 @@ pub type Moment = u64;
 /// Credential type
 pub type Credential = u64;
 
+/// Represents a thread identifier for both Forum and Proposals Discussion
```
Comment (bedeho) [inline runtime/src/lib.rs:97]: Can you here add an extended comment explaining the nature of the problem, and thus why the values MUST indeed be the same currently. That context is lost in the source.

**Review by bedeho — CHANGES_REQUESTED** [review]
I think this is fine for now, but really Polkadotjs needs to be fixed, rather than us having an opaque constraint in our runtime.

File: runtime/src/lib.rs (hunk 1 above)
Comment (mnaamani) [reply runtime/src/lib.rs:97]: done in de4ce1ef37309d96f748358ca429e3667b7176d2

**Pull request merged**